redfish-core/lib/processor.hpp

```diff
@@ -975,6 +995,280 @@ inline void
+                      const std::string& processorId, Handler&& handler)
+{
+    constexpr std::array<std::string_view, 1> interfaces = {
+        "xyz.openbmc_project.Inventory.Item.Cpu"};
+    dbus::utility::getSubTreePaths(
+        "/xyz/openbmc_project/inventory", 0, interfaces,
+        [asyncResp, processorId, handler = std::forward<Handler>(handler)](
+            const boost::system::error_code& ec,
+            const std::vector<std::string>& subTreePaths) {
+        handleProcessorPaths(asyncResp, processorId, handler, ec, subTreePaths);
+    });
+}
+
+inline void afterGetCpuCoreDataByService(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& objPath, const boost::system::error_code& ec,
+    const dbus::utility::ManagedObjectType& dbusData)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error, ec: {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    for (const auto& [path, interfaces] : dbusData)
+    {
+        if (path != objPath)
+        {
+            continue;
+        }
+
+        bool present = true;
+        bool functional = true;
+
+        for (const auto& [interface, properties] : interfaces)
+        {
+            if (interface ==
+                "xyz.openbmc_project.State.Decorator.OperationalStatus")
+            {
+                for (const auto& [proName, proValue] : properties)
+                {
+                    if (proName == "Functional")
+                    {
+                        const bool* value = std::get_if<bool>(&proValue);
+                        if (value == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        functional = *value;
+                    }
+                }
+            }
+            else if (interface == "xyz.openbmc_project.Inventory.Item")
+            {
+                for (const auto& [proName, proValue] : properties)
+                {
+                    if (proName == "Present")
+                    {
+                        const bool* value = std::get_if<bool>(&proValue);
+                        if (value == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        present = *value;
+                    }
+                    else if (proName == "PrettyName")
+                    {
+                        const std::string* prettyName =
+                            std::get_if<std::string>(&proValue);
+                        if (prettyName == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        asyncResp->res.jsonValue["Name"] = *prettyName;
```

Comment:
L1108-L1117, even if it is also done this in 1060, should we use `getPrettyName()` to make it consistent?

Reply:
Good point, I've made this change. It does result in another D-Bus call to get the PrettyName value since getPrettyName() always does the call.